# HG changeset patch
# User coleenp
# Date 1396636676 18000
#      Fri Apr 04 13:37:56 2014 -0500
# Node ID 02f12a9d5aec4489bef45e82ddcbd1d644a54dad
# Parent  ed0d828486a40609af52f45066d33f2458518f8d
8037076: Check constant pool constants
Summary: Fix events log string.
Reviewed-by: kvn, mschoene
Contributed-by: paul.nauman@oracle.com

diff -r ed0d828486a4 -r 02f12a9d5aec src/share/vm/utilities/events.cpp
--- openjdk/hotspot/src/share/vm/utilities/events.cpp	Thu Apr 03 11:19:47 2014 -0700
+++ openjdk/hotspot/src/share/vm/utilities/events.cpp	Fri Apr 04 13:37:56 2014 -0500
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 1997, 2012, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 1997, 2014, Oracle and/or its affiliates. All rights reserved.
  * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
  *
  * This code is free software; you can redistribute it and/or modify it
@@ -93,7 +93,7 @@
     va_start(ap, format);
     // Save a copy of begin message and log it.
     _buffer.printv(format, ap);
-    Events::log(NULL, _buffer);
+    Events::log(NULL, "%s", (const char*)_buffer);
     va_end(ap);
   }
 }
@@ -102,6 +102,6 @@
   if (LogEvents) {
     // Append " done" to the begin message and log it
     _buffer.append(" done");
-    Events::log(NULL, _buffer);
+    Events::log(NULL, "%s", (const char*)_buffer);
   }
 }
